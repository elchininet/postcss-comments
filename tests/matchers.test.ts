import fs from 'fs';
import postcss from 'postcss';
import postcssComments from '../src';

export const readCSSFile = (): Promise<string> => new Promise((resolve) => {
    fs.readFile(`${__dirname}/input.css`, 'utf8', (error, data): void => {
        if (error) {
            resolve('');
        } else {
            resolve(data);
        }
    });
});

describe('Matchers', () => {

    let input = '';
  
    beforeEach(async (): Promise<void> => {
      input = input || await readCSSFile();
    });

    it('String matchers', () => {
        const rulesMatchers = [
            {
                matcher: '.test18::after',
                prepend: 'Comment before test18::after'
            },
            {
                matcher: '.test2',
                append: 'Comment after .test2'
            },
            {
                matcher: '.test46',
                prepend: 'Comment inside media-query prepend',
                append: 'Comment inside media-query append'
            }
        ];
        const output = postcss([postcssComments({ rulesMatchers })]).process(input);
        expect(output.css).toMatchSnapshot();
        expect(output.warnings()).toHaveLength(0);
    });

    it('Regexp matchers', () => {
        const rulesMatchers = [
            {
                matcher: /^\.test\d+$/,
                prepend: 'Comment before any class with .test and a number'
            },
            {
                matcher: /^\.test\w+\d+$/,
                append: 'Comment after any class with .test some letters and a number'
            },
            {
                matcher: /^\.test[\s\S]+\.test\d+$/,
                prepend: 'Comment before multiple tests class rules',
                append: 'Comment after multiple test class rules'
            }
        ];
        const output = postcss([postcssComments({ rulesMatchers })]).process(input);
        expect(output.css).toMatchSnapshot();
        expect(output.warnings()).toHaveLength(0);
    });

    it('Array of matchers', () => {
        const rulesMatchers = [
            {
                matcher: [
                    /^\.test\d+-\w+$/,
                    '.example35',
                    /html \.test\d+$/
                ],
                prepend: 'Comment before these matchers'
            },
            {
                matcher: [
                    /^\.test\d+$/,
                    /^#test.+$/
                ],
                prepend: 'Start of a test',
                append: 'End of a test'
            }
        ];
        const output = postcss([postcssComments({ rulesMatchers })]).process(input);
        expect(output.css).toMatchSnapshot();
        expect(output.warnings()).toHaveLength(0);
    });

    it('Error no comments', () => {
        expect(
            // @ts-ignore
            () => postcss([postcssComments()]).process(input)
        ).toThrow();
        expect(
            () => postcss([postcssComments({ rulesMatchers: [] })]).process(input)
        ).toThrow();
    });
});