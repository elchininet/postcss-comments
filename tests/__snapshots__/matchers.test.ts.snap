// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Matchers Array of matchers 1`] = `
".test1, .test2 {
    background: url("/folder/subfolder/icons/ltr/chevron-left.png");
    background-color: #FFF;
    background-position: 10px 20px;
    border-radius: 0 2px 0 8px;
    color: #666;
    padding-right: 20px;
    text-align: left;
    transform: translate(-50%, 50%);
    width: 100%;
}

/* Start of a test */
.test2 {
    color: red;
    text-align: left;
    text-align: center;
}
/* End of a test */

/* A comment */
/* Start of a test */
.test3 {
    direction: ltr;
    margin: 1px 2px 3px;
    padding: 10px 20px;
    text-align: center;
}
/* End of a test */

/* Start of a test */
.test4 {
    font-size: 10px;
    border-color: red;
    border-radius: 2px 4px 8px 16px;
    text-shadow: red 99px -1px 1px, blue 99px 2px 1px;
    transform-origin: 10px 20px;
    transform: scale(0.5, 0.5);
}
/* End of a test */

.test5,
.test6,
.test7 {
    background: linear-gradient(0.25turn, #3F87A6, #EBF8E1, #F69D3C);
    border: 1px solid 2px;
    border-color: #666 #777 #888 #999;
    box-sizing: border-box;
    border-width: 1px 2px 3px 4px;
    position: absolute;
    left: 100px;
    transform: translateX(5px);
}

/* Start of a test */
.test8 {
    background: linear-gradient(to left, #333, #333 50%, #EEE 75%, #333 75%);
    display: flex;
    padding-left: 10%;
}
/* End of a test */

.test9, .test10 {
    background: linear-gradient(217deg, rgba(255,0,0,.8), rgba(255,0,0,0) 70.71%),
                linear-gradient(127deg, rgba(0,255,0,.8), rgba(0,255,0,0) 70.71%),
                linear-gradient(336deg, rgba(0,0,255,.8), rgba(0,0,255,0) 70.71%);
    left: 5px;
    padding: 0px 2px 3px 4px;
}

.test11:hover,
.test11:active {
    font-family: Arial, Helvetica;
    font-size: 20px;
    color: '#FFF';
    transform: translateY(10px);
    padding: 10px;
}

/* Start of a test */
#test12, #test13 {
    left: 10px;
    position: relative;
    text-align: right;
}
/* End of a test */

.test14 .test15 span {
    border-left-color: #777;
    margin: 10px 20px 30px 40px;
    transform: translate(100%, 10%);
}

/* Start of a test */
.test16 {
    padding-right: 10px;
}
/* End of a test */

.test16:hover {
    padding-right: 20px;
}

/* Start of a test */
.test17 {
    cursor: pointer;
    padding: 10px 20px 40px 10px;
    text-align: right;
}
/* End of a test */

@media only screen and (min-device-width: 320px) {
    /* Start of a test */
    .test17 {
        cursor: wait;
    }
    /* End of a test */
}

/* Start of a test */
.test18 {
    animation: 5s flip 1s ease-in-out,
               3s my-animation 6s ease-in-out;
    font-size: 10px;
    padding: 10px 20px 40px 10px;
}
/* End of a test */

.test18::after {
    content: '';
    left: 10px;
    text-align: left;
    padding-left: 5px;
    margin-left: 15px;
    transform: translateX(5px);
}

@keyframes flip {
    from {
        transform: translateX(100px);
    }
    to {
        transform: translateX(0);
    }
}

@media only screen and (min-device-width: 320px) {
    /* Start of a test */
    .test18 {
        padding: 1px 2px 3px 4px;
        width: 100%;
    }
    /* End of a test */
}

/* Start of a test */
.test19 {
    animation-delay: 1s;
    animation-duration: 3s;
    animation-name: my-animation;
    animation-timing-function: ease-in-out;
}
/* End of a test */

/* Start of a test */
.test20 {
    animation-delay: 2s;
    animation-duration: 4s;
    animation-name: my-animation, no-flip;
    animation-timing-function: ease;
}
/* End of a test */

@keyframes my-animation {
    from {
        left: 0%;
    }
    to {
        left: 100%;
    }
}

/* Start of a test */
.test21 {
    animation-delay: 1s;
    animation-duration: 3s;
    animation-name: no-flip;
    animation-timing-function: ease-in-out;
    width: 100%;
}
/* End of a test */

@keyframes no-flip {
    from {
        color: #000;
    }
    to {
        color: #FFF;
    }
}

/* Start of a test */
.test22 {
    left: 5px;
    right: 10px;
}
/* End of a test */

/* Start of a test */
.test23 {
    left: 10px;
    right: 10px;
}
/* End of a test */

/* Start of a test */
.test24 {
    border: 1px solid #FFF;
    border-bottom-color: #666;
    padding: 10px;
}
/* End of a test */

/* Comment before these matchers */
.test25-ltr {
    box-sizing: border-box;
    color: #FFF;
    font-size: 10px;
    width: 100%;
}

.test25, .test26-ltr, .test27 {
    background-image: url("/icons/icon-l.png")
}

/* Comment before these matchers */
.test26-rtl {
    background-image: url("/icons/icon-r.png")
}

/* Comment before these matchers */
.test27-prev {
    background-image: url("/icons/icon-p.png")
}

/* Comment before these matchers */
.test27-next {
    background-image: url("/icons/icon-n.png")
}

/* Start of a test */
.test28 {
    font-family: 'Material Icons';
    font-weight: normal;
    font-style: normal;
    font-size: 24px;
    display: inline-block;
    line-height: 1;
    text-transform: none;
    letter-spacing: normal;
    word-wrap: normal;
    white-space: nowrap;
}
/* End of a test */

.test28-left::before {
    content: "keyboard_arrow_left";
}

.test28-right::before {
    content: "keyboard_arrow_right";
}

.testleft29 {
    border: 1px solid gray;
}

.testleft30 {
    content: "keyboard_arrow_left";
}

.testright30 {
    content: "keyboard_arrow_right";
}

/* Start of a test */
.test31 {
    background-image: url("/icons/icon-left.png");
    border: 1px solid gray;
}
/* End of a test */

/* Start of a test */
.test32 {
    align-items: center;
    background-image: url("/icons/icon-left.png");
    background-repeat: no-repeat;
    border: 1px solid gray;    
}
/* End of a test */

/* Start of a test */
.test33 {
    color: #EFEFEF;
    left: 10px;
}
/* End of a test */

.example34 {
    color: #EFEFEF;
    left: 10px;
    width: 100%;    
}

/* Comment before these matchers */
.example35 {
    transform: translate(10px, 20px);
}

/* Start of a test */
.test36 {
    color: #FFF;
    border-left: 1px solid #666;
    padding: 10px 5px 10px 20px;
    text-align: left;
    width: 100%;
}
/* End of a test */

/* Start of a test */
.test37 {
    color: #FFF;
    border-left: 1px solid #666;
    padding: 10px 5px 10px 20px;
    text-align: left;
    width: 100%;
}
/* End of a test */

/* Start of a test */
.test38 {
    color: #FFF;
    border-left: 1px solid #666;
    padding: 10px 5px 10px 20px;
    text-align: left;
    width: 100%;
}
/* End of a test */

/* Start of a test */
.test39 {
    margin-left: 10px;
    width: 50%;
}
/* End of a test */

/* Start of a test */
.test40 {
    color: transparent;
    padding: 10px;
    left: 5px;
}
/* End of a test */

/* Start of a test */
.test41 {
    color: #EFEFEF;
    left: 10px;
}
/* End of a test */

/* Start of a test */
.test42 {
    color: #EFEFEF;
    left: 10px;
    width: 100%;    
}
/* End of a test */

/* Start of a test */
.test43 {
    transform: translate(10px, 20px);
}
/* End of a test */

@keyframes normalFlip {
    from {
        left: 0px;
        top: 0px;
    }
    to {
        left: 100px;
        top: -100px;
    }
}

/* Start of a test */
.test44 {
    animation: 5s normalFlip 1s ease-in-out;
}
/* End of a test */

@keyframes inversedFlip {
    from {
        left: 0px;
        top: 0px;
    }
    to {
        left: 100px;
        top: -100px;
    }
}

/* Start of a test */
.test45 {
    animation: 5s inversedFlip 1s ease-in-out;
}
/* End of a test */

@media only screen and (min-device-width: 320px) {
    /* Start of a test */
    .test46 {
        cursor: wait;
        text-align: left;
    }
    /* End of a test */
    /* Start of a test */
    .test47 {
        color: white;
    }
    /* End of a test */
}

@media only screen and (min-device-width: 320px) {
    /* Start of a test */
    .test48 {
        cursor: wait;
        text-align: left;
    }
    /* End of a test */
    /* Start of a test */
    .test49 {
        color: white;
    }
    /* End of a test */
}

:root {
    text-align: left;
}

/* Comment before these matchers */
html .test50 {
    color: red;
    left: 10px;
}

/* Start of a test */
.test51 {
    border-left: 1px solid #FFF;
    border: none;
}
/* End of a test */"
`;

exports[`Matchers Regexp matchers 1`] = `
"/* Comment before multiple tests class rules */
.test1, .test2 {
    background: url("/folder/subfolder/icons/ltr/chevron-left.png");
    background-color: #FFF;
    background-position: 10px 20px;
    border-radius: 0 2px 0 8px;
    color: #666;
    padding-right: 20px;
    text-align: left;
    transform: translate(-50%, 50%);
    width: 100%;
}
/* Comment after multiple test class rules */

/* Comment before any class with .test and a number */
.test2 {
    color: red;
    text-align: left;
    text-align: center;
}

/* A comment */
/* Comment before any class with .test and a number */
.test3 {
    direction: ltr;
    margin: 1px 2px 3px;
    padding: 10px 20px;
    text-align: center;
}

/* Comment before any class with .test and a number */
.test4 {
    font-size: 10px;
    border-color: red;
    border-radius: 2px 4px 8px 16px;
    text-shadow: red 99px -1px 1px, blue 99px 2px 1px;
    transform-origin: 10px 20px;
    transform: scale(0.5, 0.5);
}

/* Comment before multiple tests class rules */
.test5,
.test6,
.test7 {
    background: linear-gradient(0.25turn, #3F87A6, #EBF8E1, #F69D3C);
    border: 1px solid 2px;
    border-color: #666 #777 #888 #999;
    box-sizing: border-box;
    border-width: 1px 2px 3px 4px;
    position: absolute;
    left: 100px;
    transform: translateX(5px);
}
/* Comment after multiple test class rules */

/* Comment before any class with .test and a number */
.test8 {
    background: linear-gradient(to left, #333, #333 50%, #EEE 75%, #333 75%);
    display: flex;
    padding-left: 10%;
}

/* Comment before multiple tests class rules */
.test9, .test10 {
    background: linear-gradient(217deg, rgba(255,0,0,.8), rgba(255,0,0,0) 70.71%),
                linear-gradient(127deg, rgba(0,255,0,.8), rgba(0,255,0,0) 70.71%),
                linear-gradient(336deg, rgba(0,0,255,.8), rgba(0,0,255,0) 70.71%);
    left: 5px;
    padding: 0px 2px 3px 4px;
}
/* Comment after multiple test class rules */

.test11:hover,
.test11:active {
    font-family: Arial, Helvetica;
    font-size: 20px;
    color: '#FFF';
    transform: translateY(10px);
    padding: 10px;
}

#test12, #test13 {
    left: 10px;
    position: relative;
    text-align: right;
}

.test14 .test15 span {
    border-left-color: #777;
    margin: 10px 20px 30px 40px;
    transform: translate(100%, 10%);
}

/* Comment before any class with .test and a number */
.test16 {
    padding-right: 10px;
}

.test16:hover {
    padding-right: 20px;
}

/* Comment before any class with .test and a number */
.test17 {
    cursor: pointer;
    padding: 10px 20px 40px 10px;
    text-align: right;
}

@media only screen and (min-device-width: 320px) {
    /* Comment before any class with .test and a number */
    .test17 {
        cursor: wait;
    }
}

/* Comment before any class with .test and a number */
.test18 {
    animation: 5s flip 1s ease-in-out,
               3s my-animation 6s ease-in-out;
    font-size: 10px;
    padding: 10px 20px 40px 10px;
}

.test18::after {
    content: '';
    left: 10px;
    text-align: left;
    padding-left: 5px;
    margin-left: 15px;
    transform: translateX(5px);
}

@keyframes flip {
    from {
        transform: translateX(100px);
    }
    to {
        transform: translateX(0);
    }
}

@media only screen and (min-device-width: 320px) {
    /* Comment before any class with .test and a number */
    .test18 {
        padding: 1px 2px 3px 4px;
        width: 100%;
    }
}

/* Comment before any class with .test and a number */
.test19 {
    animation-delay: 1s;
    animation-duration: 3s;
    animation-name: my-animation;
    animation-timing-function: ease-in-out;
}

/* Comment before any class with .test and a number */
.test20 {
    animation-delay: 2s;
    animation-duration: 4s;
    animation-name: my-animation, no-flip;
    animation-timing-function: ease;
}

@keyframes my-animation {
    from {
        left: 0%;
    }
    to {
        left: 100%;
    }
}

/* Comment before any class with .test and a number */
.test21 {
    animation-delay: 1s;
    animation-duration: 3s;
    animation-name: no-flip;
    animation-timing-function: ease-in-out;
    width: 100%;
}

@keyframes no-flip {
    from {
        color: #000;
    }
    to {
        color: #FFF;
    }
}

/* Comment before any class with .test and a number */
.test22 {
    left: 5px;
    right: 10px;
}

/* Comment before any class with .test and a number */
.test23 {
    left: 10px;
    right: 10px;
}

/* Comment before any class with .test and a number */
.test24 {
    border: 1px solid #FFF;
    border-bottom-color: #666;
    padding: 10px;
}

.test25-ltr {
    box-sizing: border-box;
    color: #FFF;
    font-size: 10px;
    width: 100%;
}

/* Comment before multiple tests class rules */
.test25, .test26-ltr, .test27 {
    background-image: url("/icons/icon-l.png")
}
/* Comment after multiple test class rules */

.test26-rtl {
    background-image: url("/icons/icon-r.png")
}

.test27-prev {
    background-image: url("/icons/icon-p.png")
}

.test27-next {
    background-image: url("/icons/icon-n.png")
}

/* Comment before any class with .test and a number */
.test28 {
    font-family: 'Material Icons';
    font-weight: normal;
    font-style: normal;
    font-size: 24px;
    display: inline-block;
    line-height: 1;
    text-transform: none;
    letter-spacing: normal;
    word-wrap: normal;
    white-space: nowrap;
}

.test28-left::before {
    content: "keyboard_arrow_left";
}

.test28-right::before {
    content: "keyboard_arrow_right";
}

.testleft29 {
    border: 1px solid gray;
}
/* Comment after any class with .test some letters and a number */

.testleft30 {
    content: "keyboard_arrow_left";
}
/* Comment after any class with .test some letters and a number */

.testright30 {
    content: "keyboard_arrow_right";
}
/* Comment after any class with .test some letters and a number */

/* Comment before any class with .test and a number */
.test31 {
    background-image: url("/icons/icon-left.png");
    border: 1px solid gray;
}

/* Comment before any class with .test and a number */
.test32 {
    align-items: center;
    background-image: url("/icons/icon-left.png");
    background-repeat: no-repeat;
    border: 1px solid gray;    
}

/* Comment before any class with .test and a number */
.test33 {
    color: #EFEFEF;
    left: 10px;
}

.example34 {
    color: #EFEFEF;
    left: 10px;
    width: 100%;    
}

.example35 {
    transform: translate(10px, 20px);
}

/* Comment before any class with .test and a number */
.test36 {
    color: #FFF;
    border-left: 1px solid #666;
    padding: 10px 5px 10px 20px;
    text-align: left;
    width: 100%;
}

/* Comment before any class with .test and a number */
.test37 {
    color: #FFF;
    border-left: 1px solid #666;
    padding: 10px 5px 10px 20px;
    text-align: left;
    width: 100%;
}

/* Comment before any class with .test and a number */
.test38 {
    color: #FFF;
    border-left: 1px solid #666;
    padding: 10px 5px 10px 20px;
    text-align: left;
    width: 100%;
}

/* Comment before any class with .test and a number */
.test39 {
    margin-left: 10px;
    width: 50%;
}

/* Comment before any class with .test and a number */
.test40 {
    color: transparent;
    padding: 10px;
    left: 5px;
}

/* Comment before any class with .test and a number */
.test41 {
    color: #EFEFEF;
    left: 10px;
}

/* Comment before any class with .test and a number */
.test42 {
    color: #EFEFEF;
    left: 10px;
    width: 100%;    
}

/* Comment before any class with .test and a number */
.test43 {
    transform: translate(10px, 20px);
}

@keyframes normalFlip {
    from {
        left: 0px;
        top: 0px;
    }
    to {
        left: 100px;
        top: -100px;
    }
}

/* Comment before any class with .test and a number */
.test44 {
    animation: 5s normalFlip 1s ease-in-out;
}

@keyframes inversedFlip {
    from {
        left: 0px;
        top: 0px;
    }
    to {
        left: 100px;
        top: -100px;
    }
}

/* Comment before any class with .test and a number */
.test45 {
    animation: 5s inversedFlip 1s ease-in-out;
}

@media only screen and (min-device-width: 320px) {
    /* Comment before any class with .test and a number */
    .test46 {
        cursor: wait;
        text-align: left;
    }
    /* Comment before any class with .test and a number */
    .test47 {
        color: white;
    }
}

@media only screen and (min-device-width: 320px) {
    /* Comment before any class with .test and a number */
    .test48 {
        cursor: wait;
        text-align: left;
    }
    /* Comment before any class with .test and a number */
    .test49 {
        color: white;
    }
}

:root {
    text-align: left;
}

html .test50 {
    color: red;
    left: 10px;
}

/* Comment before any class with .test and a number */
.test51 {
    border-left: 1px solid #FFF;
    border: none;
}"
`;

exports[`Matchers String matchers 1`] = `
".test1, .test2 {
    background: url("/folder/subfolder/icons/ltr/chevron-left.png");
    background-color: #FFF;
    background-position: 10px 20px;
    border-radius: 0 2px 0 8px;
    color: #666;
    padding-right: 20px;
    text-align: left;
    transform: translate(-50%, 50%);
    width: 100%;
}

.test2 {
    color: red;
    text-align: left;
    text-align: center;
}
/* Comment after .test2 */

/* A comment */
.test3 {
    direction: ltr;
    margin: 1px 2px 3px;
    padding: 10px 20px;
    text-align: center;
}

.test4 {
    font-size: 10px;
    border-color: red;
    border-radius: 2px 4px 8px 16px;
    text-shadow: red 99px -1px 1px, blue 99px 2px 1px;
    transform-origin: 10px 20px;
    transform: scale(0.5, 0.5);
}

.test5,
.test6,
.test7 {
    background: linear-gradient(0.25turn, #3F87A6, #EBF8E1, #F69D3C);
    border: 1px solid 2px;
    border-color: #666 #777 #888 #999;
    box-sizing: border-box;
    border-width: 1px 2px 3px 4px;
    position: absolute;
    left: 100px;
    transform: translateX(5px);
}

.test8 {
    background: linear-gradient(to left, #333, #333 50%, #EEE 75%, #333 75%);
    display: flex;
    padding-left: 10%;
}

.test9, .test10 {
    background: linear-gradient(217deg, rgba(255,0,0,.8), rgba(255,0,0,0) 70.71%),
                linear-gradient(127deg, rgba(0,255,0,.8), rgba(0,255,0,0) 70.71%),
                linear-gradient(336deg, rgba(0,0,255,.8), rgba(0,0,255,0) 70.71%);
    left: 5px;
    padding: 0px 2px 3px 4px;
}

.test11:hover,
.test11:active {
    font-family: Arial, Helvetica;
    font-size: 20px;
    color: '#FFF';
    transform: translateY(10px);
    padding: 10px;
}

#test12, #test13 {
    left: 10px;
    position: relative;
    text-align: right;
}

.test14 .test15 span {
    border-left-color: #777;
    margin: 10px 20px 30px 40px;
    transform: translate(100%, 10%);
}

.test16 {
    padding-right: 10px;
}

.test16:hover {
    padding-right: 20px;
}

.test17 {
    cursor: pointer;
    padding: 10px 20px 40px 10px;
    text-align: right;
}

@media only screen and (min-device-width: 320px) {
    .test17 {
        cursor: wait;
    }
}

.test18 {
    animation: 5s flip 1s ease-in-out,
               3s my-animation 6s ease-in-out;
    font-size: 10px;
    padding: 10px 20px 40px 10px;
}

/* Comment before test18::after */
.test18::after {
    content: '';
    left: 10px;
    text-align: left;
    padding-left: 5px;
    margin-left: 15px;
    transform: translateX(5px);
}

@keyframes flip {
    from {
        transform: translateX(100px);
    }
    to {
        transform: translateX(0);
    }
}

@media only screen and (min-device-width: 320px) {
    .test18 {
        padding: 1px 2px 3px 4px;
        width: 100%;
    }
}

.test19 {
    animation-delay: 1s;
    animation-duration: 3s;
    animation-name: my-animation;
    animation-timing-function: ease-in-out;
}

.test20 {
    animation-delay: 2s;
    animation-duration: 4s;
    animation-name: my-animation, no-flip;
    animation-timing-function: ease;
}

@keyframes my-animation {
    from {
        left: 0%;
    }
    to {
        left: 100%;
    }
}

.test21 {
    animation-delay: 1s;
    animation-duration: 3s;
    animation-name: no-flip;
    animation-timing-function: ease-in-out;
    width: 100%;
}

@keyframes no-flip {
    from {
        color: #000;
    }
    to {
        color: #FFF;
    }
}

.test22 {
    left: 5px;
    right: 10px;
}

.test23 {
    left: 10px;
    right: 10px;
}

.test24 {
    border: 1px solid #FFF;
    border-bottom-color: #666;
    padding: 10px;
}

.test25-ltr {
    box-sizing: border-box;
    color: #FFF;
    font-size: 10px;
    width: 100%;
}

.test25, .test26-ltr, .test27 {
    background-image: url("/icons/icon-l.png")
}

.test26-rtl {
    background-image: url("/icons/icon-r.png")
}

.test27-prev {
    background-image: url("/icons/icon-p.png")
}

.test27-next {
    background-image: url("/icons/icon-n.png")
}

.test28 {
    font-family: 'Material Icons';
    font-weight: normal;
    font-style: normal;
    font-size: 24px;
    display: inline-block;
    line-height: 1;
    text-transform: none;
    letter-spacing: normal;
    word-wrap: normal;
    white-space: nowrap;
}

.test28-left::before {
    content: "keyboard_arrow_left";
}

.test28-right::before {
    content: "keyboard_arrow_right";
}

.testleft29 {
    border: 1px solid gray;
}

.testleft30 {
    content: "keyboard_arrow_left";
}

.testright30 {
    content: "keyboard_arrow_right";
}

.test31 {
    background-image: url("/icons/icon-left.png");
    border: 1px solid gray;
}

.test32 {
    align-items: center;
    background-image: url("/icons/icon-left.png");
    background-repeat: no-repeat;
    border: 1px solid gray;    
}

.test33 {
    color: #EFEFEF;
    left: 10px;
}

.example34 {
    color: #EFEFEF;
    left: 10px;
    width: 100%;    
}

.example35 {
    transform: translate(10px, 20px);
}

.test36 {
    color: #FFF;
    border-left: 1px solid #666;
    padding: 10px 5px 10px 20px;
    text-align: left;
    width: 100%;
}

.test37 {
    color: #FFF;
    border-left: 1px solid #666;
    padding: 10px 5px 10px 20px;
    text-align: left;
    width: 100%;
}

.test38 {
    color: #FFF;
    border-left: 1px solid #666;
    padding: 10px 5px 10px 20px;
    text-align: left;
    width: 100%;
}

.test39 {
    margin-left: 10px;
    width: 50%;
}

.test40 {
    color: transparent;
    padding: 10px;
    left: 5px;
}

.test41 {
    color: #EFEFEF;
    left: 10px;
}

.test42 {
    color: #EFEFEF;
    left: 10px;
    width: 100%;    
}

.test43 {
    transform: translate(10px, 20px);
}

@keyframes normalFlip {
    from {
        left: 0px;
        top: 0px;
    }
    to {
        left: 100px;
        top: -100px;
    }
}

.test44 {
    animation: 5s normalFlip 1s ease-in-out;
}

@keyframes inversedFlip {
    from {
        left: 0px;
        top: 0px;
    }
    to {
        left: 100px;
        top: -100px;
    }
}

.test45 {
    animation: 5s inversedFlip 1s ease-in-out;
}

@media only screen and (min-device-width: 320px) {
    /* Comment inside media-query prepend */
    .test46 {
        cursor: wait;
        text-align: left;
    }
    /* Comment inside media-query append */
    .test47 {
        color: white;
    }
}

@media only screen and (min-device-width: 320px) {
    .test48 {
        cursor: wait;
        text-align: left;
    }
    .test49 {
        color: white;
    }
}

:root {
    text-align: left;
}

html .test50 {
    color: red;
    left: 10px;
}

.test51 {
    border-left: 1px solid #FFF;
    border: none;
}"
`;
